src/main/java/org/opentestsystem/ap/ims/service/FileUploadServiceMapper.java

```diff
@@ -16,75 +16,95 @@
+    public static final String ASL_LANGUAGE = "enu";
 
     /**
-     * Creates an {@link Attachment} instance from a {@link BrailleFile}.  It is mapping an IMS model to the common
+     * Creates an {@link Attachment} instance from a {@link AttachmentFile}.  It is mapping an IMS model to the common
      * model.
      *
-     * @param brailleFile The file details being uploaded.
+     * @param attachmentFile The file details being uploaded.
      * @return An {@link Attachment} instance.
      */
-    public Attachment newAttachment(final BrailleFile brailleFile) {
+    public Attachment newAttachment(final AttachmentFile attachmentFile) {
         final Attachment attachment = new Attachment();
-        attachment.setFileName(brailleFile.getFileName());
-        attachment.setType(brailleFile.getType());
-        attachment.setSubtype(brailleFile.getSubtype());
-        attachment.setLanguage(brailleFile.getLanguage());
+        attachment.setFileName(attachmentFile.getFileName());
+        attachment.setType(attachmentFile.getType());
+        attachment.setSubtype(attachmentFile.getSubtype());
+        attachment.setLanguage(attachmentFile.getLanguage());
         return attachment;
     }
 
     /**
-     * Using the file name an instance of {@link BrailleFile} is created.  A file name is expected in a specific format.
+     * Using the file name an instance of {@link AttachmentFile} is created.  A file name is expected in a specific format.
      * It is assumed the file name format has been validated prior to calling this method.
      * <p>
      * <p>
      * "stim_8393_enu_contracted_transcript.brf" and "stim_8393_enu_contracted.brf" are examples of valid braille file
-     * names.  The file name contains the different parts making up a {@link BrailleFile}
+     * names. A valid asl file is "item_3431_ASL_STEM.mp4".
+     * The file name contains the different parts making up a {@link AttachmentFile}
      *
-     * @param fileName The file name to parse for the braille file data.
-     * @return A {@link BrailleFile} instance.
+     * @param fileName The file name to parse for the file data.
+     * @return A {@link AttachmentFile} instance.
      */
-    public BrailleFile newBrailleFile(final String fileName) {
+    public AttachmentFile newAttachmentFile(final String fileName, AttachmentFileType attachmentFileType) {
         if (fileName == null) {
             throw new ValidationException("File name cannot be null");
         }
 
         // first split on the extension
         final String[] splitOnExtension = split(fileName, ".");
 
-        final String brailleType = equalsIgnoreCase(splitOnExtension[1], BRAILLE_TYPE_BRF.toLowerCase()) ?
-            BRAILLE_TYPE_BRF : BRAILLE_TYPE_PRN;
+        final AttachmentFile attachmentFile = new AttachmentFile();
+        attachmentFile.setFileName(fileName);
 
+        String fileType = StringUtils.EMPTY;
         final String[] splitOnName = split(splitOnExtension[0], "_");
```

Comment:
This one is my bad, I should have made "_" a constant like final String UNDERSCORE = "_".